src/index.js

```diff
@@ -0,0 +1,73 @@
+const flattener = require('flatten-obj')();
+
+// see http://stackoverflow.com/questions/19098797/fastest-way-to-flatten-un-flatten-nested-json-objects
+// http://jsfiddle.net/WSzec/14/
+/* eslint-disable no-use-before-define, no-param-reassign */
+function loop(cur, prop, result, exclusions) {
+  let isEmpty = true;
+
+  for (const p in cur) {
+    if (Object.prototype.hasOwnProperty.call(cur, p)) {
+      isEmpty = false;
+      recurse(cur[p], prop ? `${prop}.${p}` : p, result, exclusions);
+    }
+  }
+
+  if (isEmpty && Array.isArray(cur)) {
+    result[prop] = [];
+  } else if (isEmpty && !Array.isArray(cur)) {
+    result[prop] = {};
+  }
+}
+
+function recurse(cur, prop, result, exclusions) {
+  if (exclusions.includes(prop)) return;
+
+  if (Object(cur) !== cur) {
+    if (cur !== undefined) {
+      result[prop] = cur;
+    }
+  } else {
+    loop(cur, prop, result, exclusions);
+  }
+}
+
+module.exports = function (data, exclusions) {
+  const result = {};
+
+  exclusions = exclusions || [];
+
+  recurse(data, '', result, exclusions);
+  return result;
+};
+
+module.exports.camelCase = (obj) => {
```

Comment:
Ok thank you, I need to document myself a bit but i'll ask you if there are things that I have trouble understanding !